src/core/layertree/qgslayertreeutils.cpp

```diff
@@ -396,6 +416,7 @@ void QgsLayerTreeUtils::replaceChildrenOfEmbeddedGroups( QgsLayerTreeGroup *grou
       if ( child->customProperty( QStringLiteral( "embedded" ) ).toInt() )
       {
         child->setCustomProperty( QStringLiteral( "embedded-invisible-layers" ), invisibleLayerList( child ) );
+        child->setCustomProperty( QStringLiteral( "embedded-unchecked-groups" ), uncheckedGroupList( child ) );
```

Comment:
This doesn't look particularly stable -- two equally named groups will mess up the result

Reply:
Thanks for the review !

Indeed, I noticed that but I wouldn't add an id to group because it seems to me that this subject has already been talked a while ago, and furthermore switching for id group will need some refactoring I guess. But I agree this is not satisfying enough.

I also though about using the node index in each embedded group (I pushed the changes). This works too, and it accepts groups with the same name, but in case of changes in the project that is embedded (a layer/group added/removed), index may change and the checked state will be lost.

So after all group id seems to be needed. I change the PR title with WIP and I will try test to add group id

Reply:
I fixed it, adding a group's id in the tree layer !